__tests__/web/keyboard/index.spec.ts

```diff
@@ -0,0 +1,495 @@
+import { Page } from 'puppeteer/lib/types';
+import { IsolatedWorld } from '../isolatedWorld';
+
+describe('keyboard', () => {
+  let isolatedWorld: IsolatedWorld;
+  beforeEach(async () => {
+    isolatedWorld = new IsolatedWorld(page as unknown as Page);
+    await page.goto(
+      'https://dvcs.w3.org/hg/d4e/raw-file/tip/key-event-test.html',
+      {
+        waitUntil: 'domcontentloaded',
+      },
+    );
+    await isolatedWorld.injectRecorder();
+    await isolatedWorld.useFakeDateTime(100);
+    await isolatedWorld.createRecorder();
+  });
+
+  it('should match texting event', async () => {
+    await (await page.$('input'))?.focus();
+    await page.keyboard.type('12345');
+    // expect record keyboard event and match type texting;
+    const curr = await isolatedWorld.retrieveCurrentEvent();
+    expect(curr.events).toMatchInlineSnapshot(`
+      Array [
+        Object {
+          "code": "Digit1",
+          "key": "1",
+          "keyCode": 49,
+          "modifiers": Object {},
```

Comment:
modifier value is empty  matches expection ?

Reply:
becaues no modified key is pressed when typing, i will add some test with modifier soon

Reply:
should it has default value :  `{ metaKey: false, shiftKey: false ，ctrlKey: false, altKey: false }` ？

Reply:
I will update toModifier function to remove or operator